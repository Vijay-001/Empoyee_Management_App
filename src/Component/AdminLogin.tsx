import React, { useState } from 'react';
import Button from '@material-ui/core/Button';
import CssBaseline from '@material-ui/core/CssBaseline';
import Container from '@material-ui/core/Container';
import Link from '@material-ui/core/Link';
import Grid from '@material-ui/core/Grid';
import { IUser, LoginUser } from '../Reducer/userReducer';
import { useDispatch } from 'react-redux';
import { useAppSelector } from '../Reducer/reducerHooks';
import { Redirect } from 'react-router-dom';
import { FormikProps, withFormik } from "formik";
import * as Yup from "yup";
import { TextField } from '@material-ui/core';




interface FormValues {
    email: string;
    password: string;
}

interface OtherProps {
    title?: string;
    userInfo: IUser
}

interface MyFormProps {
    initialEmail?: string;
    initialPassword?: string;
    userInfo: IUser
}




const AdminLoginForm = (props: OtherProps & FormikProps<FormValues>) => {

    const {
        values,
        errors,
        touched,
        handleChange,
        handleBlur,
        handleSubmit,
        isSubmitting,
        userInfo
    } = props;


    const [userInfoState, setUserInfoState] = useState(userInfo);

    const dispatch = useDispatch();


    const onChangesValues = (fieldName: string, value: string) => {
        setUserInfoState({
            ...userInfoState,
            [fieldName]: value
        })
    }

    const CheckLogindetails = (e: React.FormEvent<HTMLFormElement>) => {
        e.preventDefault();
        handleSubmit();
        dispatch(LoginUser(userInfoState));  
            
     }

    // getting data from  reducer custom hooks

    const { data } = useAppSelector((state) => state.users.users);
    if (data[0] && typeof userInfoState !== 'undefined' && userInfoState !== null) {
        return <Redirect to='/ViewEmployeeDetails' />
    } 



    return (
        <form onSubmit={CheckLogindetails}>
            <Container component="main" maxWidth="xs">
                <CssBaseline />
                <p></p>
                <p></p>
                <p></p>
                <p></p>
                <Grid container spacing={2}>
                    <Grid item xs={12}>
                        <TextField
                            variant="outlined"
                            name="email"
                            label="Email Address"
                            size="small"
                            margin="normal"
                            onKeyUp={(event: any) => { onChangesValues('email', event.target.value) }}
                            onBlur={handleBlur}
                            value={values?.email}
                            error={Boolean(touched.email && errors.email)}                            
                            helperText={touched.email && errors.email}
                            onChange={handleChange}
                            fullWidth
                        />
                    </Grid>

                    <Grid item xs={12}>
                        <TextField
                            variant="outlined"
                            name="password"
                            label="Password"
                            margin="normal"
                            size="small"
                            onKeyUp={(event: any) => { onChangesValues('password', event.target.value) }}
                            onChange={handleChange}
                            onBlur={handleBlur}
                            value={values?.password}
                            error={Boolean(touched.password && errors.password)}                            
                            helperText={touched.password && errors.password}
                            fullWidth
                        />
                    </Grid>
                </Grid>
                <p></p>
                <p></p>
                <p></p>
                <p></p>
                <Button
                    type="submit"
                    fullWidth
                    variant="contained"
                    color="primary"
                    disabled={
                        isSubmitting ||
                        !!(errors.email && touched.email) ||
                        !!(errors.password && touched.password)
                    }
                >
                    Sign In
               </Button>
                <p></p>
                <p></p>
                <Grid container>
                    <Grid item xs>
                        <Link href="#" variant="body2">
                            Forgot password?
              </Link>
                    </Grid>
                    <p></p>
                    <p></p>
                    <Grid item>
                        <Link href="/signup" variant="body2">
                            {"Don't have an account? Sign Up"}
                        </Link>
                    </Grid>
                </Grid>
            </Container>
        </form>       
        
      )
 }


const AdminLogin = withFormik<MyFormProps, FormValues>({

    mapPropsToValues: props => ({
        email: props.initialEmail || "",
        password: props.initialPassword || ""
    }),

    validationSchema: Yup.object().shape({
        email: Yup.string().email("Email not valid").required("Email is required"),
        password: Yup.string().required("Password is required")
    }),

    handleSubmit(
        { email, password }: FormValues,
        { props, setSubmitting, setErrors }
    ) {
        setSubmitting(false);
        console.log(email, password);
    }

})(AdminLoginForm);

export default AdminLogin;




/*
interface loginData {

    userInfo: IUser
}

interface FormValues {
    email: string;
    password: string;
}


// the Formik component supports yup validation out-of-the-box via the `validationSchema` prop

const validationSchema = yup.object().shape({

    email: yup.string().required("Required"),

    password: yup.string().required("Required")

});


const AdminLogin: React.FC<loginData> = props => {

     const { userInfo } = props;

    const [userInfoState, setUserInfoState] = useState(userInfo);

    const dispatch = useDispatch();

     
    const onChangesValues = (fieldName: string, value: string) => {
        setUserInfoState({
            ...userInfoState,
            [fieldName]:value
        })
    }


    const CheckLogindetails = () => {
        if (typeof userInfoState !== 'undefined' && userInfoState !== null) {        
            dispatch(LoginUser(userInfoState));
          }         
      }

   // getting data from  reducer custom hooks

    const { data } = useAppSelector((state) => state.users.users);
    if (data[0]) {
        return <Redirect to='/ViewEmployeeDetails' />
    }

             
         return (
             <Container component="main" maxWidth="xs">
                 <CssBaseline />
                 <p></p>
                 <p></p>
                 <p></p>
                 <p></p>
                 <Formik
                     initialValues={{
                         email: "",
                         password: ""
                     }}
                     validationSchema={validationSchema}
                     onSubmit={(
                         values: FormValues,
                         formikHelpers: FormikHelpers<FormValues>
                     ) => {
                           formikHelpers.setSubmitting(false);
                     }}
                 >
                     {(formikProps: FormikProps<FormValues>) => (
                         <Form noValidate autoComplete="off">

                             <Grid container spacing={2}>
                                 <Grid item xs={12}>
                                     <Field
                                         name="email"
                                         label="Email Address"
                                         size="small"
                                         component={FormTextField}
                                         onKeyUp={(event: any) => { onChangesValues('email', event.target.value) }}
                                     />


                                 </Grid>
                                 <Grid item xs={12}>
                                     <Field
                                         name="password"
                                         label="Password"
                                         size="small"
                                         component={FormTextField}
                                         onKeyUp={(event: any) => { onChangesValues('password', event.target.value) }}
                                     />

                                 </Grid>
                             </Grid>
                 <p></p>
                 <p></p>
                 <p></p>
                 <p></p>
                 <Button
                     type="submit"
                     fullWidth
                     variant="contained"
                     color="primary"
                     onClick={CheckLogindetails}
                 >
                     Sign In
               </Button>
                 <p></p>
                 <p></p>
                 <Grid container>
                     <Grid item xs>
                         <Link href="#" variant="body2">
                             Forgot password?
              </Link>
                     </Grid>
                     <p></p>
                     <p></p>
                     <Grid item>
                         <Link href="/signup" variant="body2">
                             {"Don't have an account? Sign Up"}
                         </Link>
                     </Grid>
                             </Grid>
                         </Form>
                     )}
                 </Formik>
             </Container>
         )
}

*/
/*export default AdminLogin; */
