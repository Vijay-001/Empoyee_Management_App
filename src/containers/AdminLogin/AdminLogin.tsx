import React, { useState } from 'react';
import Button from '@material-ui/core/Button';
import CssBaseline from '@material-ui/core/CssBaseline';
import Container from '@material-ui/core/Container';
import Grid from '@material-ui/core/Grid';
import { useDispatch } from 'react-redux';
import { useHistory } from 'react-router-dom';
import { FormikProps } from 'formik';
import { TextField } from '@material-ui/core';
import { adminLogin } from '../../common/Api/Api';
import { FormValues, OtherProps } from '../../common/Interface/Interface';
import { useAppSelector } from '../../store/reducers/reducerHooks';
import { FormValidation } from '../../common/Formik/Formik';

const AdminLoginForm = (props: OtherProps & FormikProps<FormValues>) => {
  const {
    values,
    errors,
    touched,
    handleChange,
    handleBlur,
    handleSubmit,
    isSubmitting,
    userInfo,
  } = props;

  const [userInfoState, setUserInfoState] = useState(userInfo);

  const dispatch = useDispatch();
  const history = useHistory();

  const onChangesValues = (fieldName: string, value: string) => {
    setUserInfoState({
      ...userInfoState,
      [fieldName]: value,
    });
  };
  const data = useAppSelector((state) => state.users.users);

  const CheckLogindetails = async (e: React.FormEvent<HTMLFormElement>) => {
    e.preventDefault();
    handleSubmit();
    if (typeof userInfoState !== 'undefined'
          && ('email' in userInfoState)
        && ('password' in userInfoState)) {
      await dispatch(adminLogin(userInfoState));
      console.log('data', data);
      history.push('/ViewEmployee');
    }
  };

  return (
    <form onSubmit={CheckLogindetails}>
      <Container component="main" maxWidth="xs">
        <CssBaseline />
        <p />
        <p />
        <p />
        <p />
        <Grid container spacing={2}>
          <Grid item xs={12}>
            <TextField
              variant="outlined"
              name="email"
              label="Email Address"
              size="small"
              margin="normal"
              onKeyUp={(event: any) => { onChangesValues('email', event.target.value); }}
              onBlur={handleBlur}
              value={values?.email}
              error={Boolean(touched.email && errors.email)}
              helperText={touched.email && errors.email}
              onChange={handleChange}
              fullWidth
            />
          </Grid>

          <Grid item xs={12}>
            <TextField
              variant="outlined"
              name="password"
              placeholder="password"
              margin="normal"
              size="small"
              type="password"
              onKeyUp={(event: any) => { onChangesValues('password', event.target.value); }}
              onChange={handleChange}
              onBlur={handleBlur}
              value={values?.password}
              error={Boolean(touched.password && errors.password)}
              helperText={touched.password && errors.password}
              fullWidth
            />
          </Grid>
        </Grid>
        <p />
        <p />
        <p />
        <p />
        <Button
          type="submit"
          fullWidth
          variant="contained"
          color="primary"
          disabled={
                        isSubmitting
                        || !!(errors.email && touched.email)
                        || !!(errors.password && touched.password)
                    }
        >
          Sign In
        </Button>
      </Container>
    </form>
  );
};

export default (FormValidation)(AdminLoginForm);
