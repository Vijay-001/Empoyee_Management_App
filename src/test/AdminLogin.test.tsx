import configureMockStore from "redux-mock-store";
import { Provider } from "react-redux";
import { fireEvent, render } from "@testing-library/react";
import AdminLogin from "../containers/userLogin/login";

const mockStore = configureMockStore();
const store = mockStore({});

describe("<AdminLogin/>", () => {
  const fieldprops = {
    first_name: "",
    id: "",
    last_name: "",
    email: "",
    avatar: "",
    password: "",
  };

  const LoginComponent = () => (
    <Provider store={store}>
      <AdminLogin userInfo={fieldprops} />
    </Provider>
  );

  it("renders default state values", () => {
    const { getByTestId } = render(<LoginComponent />);
    const password = getByTestId("userAccountPassword") as HTMLInputElement;
    const email = getByTestId("userAccountEmail") as HTMLInputElement;
    const submit = getByTestId("submitUserDetails");
    fireEvent.click(submit);
    expect(password.value).toBe("");
    expect(email.value).toBe("");
    expect(submit).toHaveClass("Mui-disabled");
  });

  it("disabled submit for invalid emailId and password", () => {
    const { getByTestId } = render(<LoginComponent />);
    const email = getByTestId("userAccountEmail");
    const password = getByTestId("userAccountPassword");
    const button = getByTestId("submitUserDetails");
    fireEvent.keyUp(email, { target: { value: "vkooogmail" } });
    fireEvent.keyUp(password, { target: { value: "data100" } });
    expect(button).not.toHaveClass("Mui-disabled");
  });
});
