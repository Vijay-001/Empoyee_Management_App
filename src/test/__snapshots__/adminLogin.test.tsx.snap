// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdminLogin/> renders correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <main
    className="MuiContainer-root MuiContainer-maxWidthXs"
  >
    <p />
    <p />
    <p />
    <p />
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
          onKeyUp={[Function]}
        >
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
              data-testid="userAccountEmail"
              disabled={false}
              name="email"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="Email Address"
              required={false}
              type="text"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
              style={
                Object {
                  "paddingLeft": 8,
                }
              }
            >
              <legend
                className="PrivateNotchedOutline-legend-2"
                style={
                  Object {
                    "width": 0.01,
                  }
                }
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
          onKeyUp={[Function]}
        >
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
              data-testid="userAccountPassword"
              disabled={false}
              name="password"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="Password"
              required={false}
              type="password"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
              style={
                Object {
                  "paddingLeft": 8,
                }
              }
            >
              <legend
                className="PrivateNotchedOutline-legend-2"
                style={
                  Object {
                    "width": 0.01,
                  }
                }
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
    <p />
    <p />
    <p />
    <p />
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
      data-testid="submitUserDetails"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="submit"
    >
      <span
        className="MuiButton-label"
      >
         
        Sign In
      </span>
    </button>
  </main>
</form>
`;

exports[`<AdminLogin/> renders correctly 2`] = `
<form
  onSubmit={[Function]}
>
  <main
    className="MuiContainer-root MuiContainer-maxWidthXs"
  >
    <p />
    <p />
    <p />
    <p />
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
          onKeyUp={[Function]}
        >
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
              data-testid="userAccountEmail"
              disabled={false}
              name="email"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="Email Address"
              required={false}
              type="text"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
              style={
                Object {
                  "paddingLeft": 8,
                }
              }
            >
              <legend
                className="PrivateNotchedOutline-legend-2"
                style={
                  Object {
                    "width": 0.01,
                  }
                }
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
          onKeyUp={[Function]}
        >
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
              data-testid="userAccountPassword"
              disabled={false}
              name="password"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="Password"
              required={false}
              type="password"
              value=""
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
              style={
                Object {
                  "paddingLeft": 8,
                }
              }
            >
              <legend
                className="PrivateNotchedOutline-legend-2"
                style={
                  Object {
                    "width": 0.01,
                  }
                }
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
    <p />
    <p />
    <p />
    <p />
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
      data-testid="submitUserDetails"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="submit"
    >
      <span
        className="MuiButton-label"
      >
         
        Sign In
      </span>
    </button>
  </main>
</form>
`;
